core/src/main/java/com/graphhopper/routing/weighting/custom/EdgeToValueEntry.java

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.weighting.custom;
+
+import com.graphhopper.util.EdgeIteratorState;
+
+/**
+ * This internal interface calculates a double value from the specified EdgeIteratorState. Used for CustomWeighting.
+ *
+ * @see {{@link GeoToValueEntry}} or {@link EnumToValueEntry}
+ */
+interface EdgeToValueEntry {
+    /**
+     * @return a double value or null if this entry should be skipped
```

Comment:
I do not understand this comment. What does 'should be skipped' mean here? And we cannot return null because the return type is a primitive (or did you mean zero)?

Reply:
Indeed, now we cannot return null ... but at one point in the past of this PR we could - I swear ;)

I think now we just return the default that leaves the value unchanged. The reason I did this was to 1. get rid of the usage of non-primitive number and 2. avoid branching like `if (value == null)`